File: rabbitmq/consumer.py
#!/usr/bin/env python3
import pika


def consumer(channel, connection):
    channel.basic_consume('foo', on_message_callback, auto_ack=True)
    channel.start_consuming()
    connection.close()


def on_message_callback(channel, method, properties, body):
    print(body)


parameters = pika.URLParameters('amqp://guest:guest@localhost:5672/%2F')
connection = pika.BlockingConnection(parameters)
chan = connection.channel()
chan.queue_declare(queue='foo')
consumer(chan, connection)

File: rabbitmq/consumer.pl
#!/usr/bin/perl
#
use Net::AMQP::RabbitMQ;
my $mq = Net::AMQP::RabbitMQ->new();
$mq->connect("localhost", { user => "guest", password => "guest" });
$mq->channel_open(1);
$mq->queue_declare(1, "foo");


while(1) {
    my $gotten = $mq->get(1, "foo");
    print $gotten->{body} . "\n";
}
$mq->disconnect();
